### PR TITLE
More code style/syntax/usage clean up

### PR DESCRIPTION
**Pull request by drwill-ms**
See code comments for a list of relevant changes and reasons

### CODE REVIEW

File: sdks/csharp/adtIotHub_delete.cs [hunk 1]
```diff
@@ -19,14 +20,24 @@ namespace Samples.AdtIothub
...
         [FunctionName("DeleteDeviceInTwinFunc")]
         public static async Task Run(
             [EventHubTrigger("lifecycleevents", Connection = "EVENTHUB_CONNECTIONSTRING")] EventData[] events, ILogger log)
         {
-            var exceptions = new List<Exception>();
+            var exceptions = new List<Exception>(events.Length);
+
+            // Create Digital Twin client
+            var cred = new ManagedIdentityCredential(adtAppId);
+            var client = new DigitalTwinsClient(
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:34]: Silly to create the client in the loop when nothing changes for the client parameters inside the loop.

File: sdks/csharp/adtIotHub_delete.cs [hunk 2]
```diff
@@ -38,21 +49,22 @@ public static class DeleteDeviceInTwinFunc
                     if (opType == "deleteDeviceIdentity")
                     {
                         string deviceId = eventData.Properties["deviceId"] as string;
-                        
-                        // Create Digital Twin client
-                        var cred = new ManagedIdentityCredential(adtAppId);
-                        var client = new DigitalTwinsClient(new Uri(adtInstanceUrl), cred, new DigitalTwinsClientOptions { Transport = new HttpClientTransport(httpClient) });
 
-                        // Find twin based on the original Registration ID
-                        string regID = deviceId; // simple mapping
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:47]: Removed unnecessary extra variable

File: sdks/csharp/adtIotHub_delete.cs [hunk 3]
```diff
@@ -38,21 +49,22 @@ public static class DeleteDeviceInTwinFunc
...
-                        string dtId = await GetTwinId(client, regID, log);
-                        if (dtId != null)
+                        try
                         {
-                            await DeleteRelationships(client, dtId, log);
+                            // Find twin based on the original Registration ID
+                            BasicDigitalTwin digitalTwin = await client.GetDigitalTwinAsync<BasicDigitalTwin>(deviceId);
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:56]: Previous helper function was running a query to find a specific digital twin by its Id, which we can do a lot more efficiently with a direct 'get' call. The downside is that if the twin is not found it'll throw an exception with a 404 status code. So we just catch that and do nothing (as before).

File: sdks/csharp/adtIotHub_delete.cs [hunk 4]
```diff
@@ -38,21 +49,22 @@ public static class DeleteDeviceInTwinFunc
...
+                            // In order to delete the twin, all relationships must first be removed
+                            await DeleteRelationshipsAsync(client, digitalTwin.Id, log);
+
+                            // Delete the twin
+                            await client.DeleteDigitalTwinAsync(digitalTwin.Id, digitalTwin.ETag);
+                            log.LogInformation($"Twin '{digitalTwin.Id}' deleted in DT");
+                        }
+                        catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.NotFound)
+                        {
+                            log.LogWarning($"Twin '{deviceId} not found in DT");
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:67]: Well, let's at least log a warning that the digital twin could not be found.

File: sdks/csharp/adtIotHub_delete.cs [hunk 5]
```diff
@@ -38,21 +49,22 @@ public static class DeleteDeviceInTwinFunc
...
+                        try
                         {
-                            await DeleteRelationships(client, dtId, log);
+                            // Find twin based on the original Registration ID
+                            BasicDigitalTwin digitalTwin = await client.GetDigitalTwinAsync<BasicDigitalTwin>(deviceId);
 
-                            // Delete twin
-                            await client.DeleteDigitalTwinAsync(dtId);
-                            log.LogInformation($"Twin '{dtId}' deleted in DT");
+                            // In order to delete the twin, all relationships must first be removed
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:58]: Added a comment as to why we'll delete relationships before deleting the twin.

File: sdks/csharp/adtIotHub_delete.cs [hunk 6]
```diff
@@ -70,37 +82,23 @@ public static class DeleteDeviceInTwinFunc
...
+            await foreach (BasicRelationship relationship in relationships)
             {
-                BasicRelationship relationship = System.Text.Json.JsonSerializer.Deserialize<BasicRelationship>(relationshipJson);
-                relationshipIds.Add(relationship.Id);
+                await client.DeleteRelationshipAsync(dtId, relationshipId.Id, relationship.ETag);
+                log.LogInformation($"Twin '{dtId}' relationship '{relationship.Id}' deleted in DT");
             }
 
-            foreach (var relationshipId in relationshipIds)
+            AsyncPageable<IncomingRelationship> incomingRelationships = client.GetIncomingRelationshipsAsync(dtId);
```
Comment (drwill-ms) [inline sdks/csharp/adtIotHub_delete.cs:97]: Incoming relationships must be deleted too.

File: sdks/csharp/eventRoute_operations.cs [hunk 7]
```diff
@@ -2,36 +2,37 @@
...
 string eventFilter = "$eventType = 'DigitalTwinTelemetryMessages' or $eventType = 'DigitalTwinLifecycleNotification'";
 var er = new DigitalTwinsEventRoute("endpointName", eventFilter);
-await client.CreateOrReplaceEventRouteAsync("routeName", er);
+await client.CreateOrReplaceEventRouteAsync("routeId", er);
 // </CreateEventRoute>
 
 // ------------------ CREATE, LIST, AND DELETE EVENT ROUTE (Full sample) ---------------------
 // <FullEventRouteSample>
-private async static Task CreateEventRoute(DigitalTwinsClient client, String routeName, DigitalTwinsEventRoute er)
+private static async Task CreateEventRouteAsync(DigitalTwinsClient client, string routeId, DigitalTwinsEventRoute er)
```
Comment (drwill-ms) [inline sdks/csharp/eventRoute_operations.cs:10]: `routeName` was renamed to `routeId` at GA

**Review by baanders — APPROVED** [review]

**Pull request merged**